spacy/pipeline/tok2vec.py

```diff
@@ -62,28 +62,44 @@ def __init__(self, vocab: Vocab, model: Model, name: str = "tok2vec") -> None:
         self.vocab = vocab
         self.model = model
         self.name = name
-        self.listeners = []
+        self.listener_map = {}
         self.cfg = {}
 
-    def add_listener(self, listener: "Tok2VecListener") -> None:
+    @property
+    def listeners(self) -> List[Model]:
+        """RETURNS (List[Model]): The listener models listening to this
+        component. Usually internals.
+        """
+        return [m for c in self.listening_components for m in self.listener_map[c]]
+
+    @property
+    def listening_components(self) -> List[str]:
+        """RETURNS (List[str]): The downstream components listening to this
+        component. Usually internals.
+        """
+        return list(self.listener_map.keys())
+
+    def add_listener(self, listener: "Tok2VecListener", component: str) -> None:
```

Comment:
I think people would pass in the actual component if we call this only `component` (especially since we have a `component` argument in another method). Perhaps `component_name`?